### PR TITLE
Add class balancing to standard recipe

### PR DESCRIPTION
**Pull request by elronbandel**
The balancing is at the "target" level after rendering so every type of class or answer can be balanced

### CODE REVIEW

File: src/unitxt/operators.py [hunk 1]
```diff
@@ -734,3 +735,52 @@ def process(self, instance: Dict[str, Any], stream_name: str = None) -> Dict[str
...
+
+    def process(self, stream: Stream, stream_name: str = None) -> Generator:
+        counter = collections.Counter()
+
+        for instance in stream:
+            counter[self.signature(instance)] += 1
+
+        lowest_count = counter.most_common()[-1][-1]
+
+        max_total_instances = lowest_count
```
Comment (yoavkatz) [inline src/unitxt/operators.py:776]: max_total_instance_per_signiture

**Review by yoavkatz — COMMENTED** [review]
Should look like this in the API:

card=wnli,train_balancer=operator.balance.determistics,max_train_instances=50,max_test_instance=100

File: tests/test_operators.py [hunk 2]
```diff
@@ -328,3 +329,22 @@ def test_take_by_field(self):
             targets=targets,
             tester=self,
         )
+
+    def test_detirministic_balancer(self):
```
Comment (yoavkatz) [inline tests/test_operators.py:333]: deterministic

File: src/unitxt/operators.py [hunk 3]
```diff
@@ -734,3 +735,60 @@ def process(self, instance: Dict[str, Any], stream_name: str = None) -> Dict[str
             dict_set(instance, field, new_values, use_dpath=True, set_multiple=True)
 
         return instance
+
+
+class StreamRefiner(SingleStreamOperator):
+    max_instances: int = None
```
Comment (yoavkatz) [inline src/unitxt/operators.py:741]: Why is this a class member? should it be a local variable?

Comment (elronbandel) [reply src/unitxt/operators.py:741]: It is to allow control over the max instances of all the streams instead of having to define max per stream

**Review by yoavkatz — APPROVED** [review]

**Pull request merged**